ivy/data_classes/array/array.py

```diff
@@ -393,6 +394,10 @@ def __getattribute__(self, item):
         return super().__getattribute__(item)
 
     def __getattr__(self, item):
+        if ivy.current_backend() != _determine_backend_from_args(self._data):
+            raise ivy.utils.exceptions.InvalidBackendException(
+                "This operation is only allowed when dynamic_backend = True"
```

Comment:
I think this message could end up being a bit misleading, as not using dynamic backend setting is one of the reasons that could lead to this issue but there's other possible reasons as well. ig an appropriate message would be something along the lines of `The current backend is {ivy.current_backend} while the ivy.Array used is based on the {array_backend} backend. Please set dynamic_backend=True for the array if you want to convert it to the target backend`

Reply:
I also agree on that as well @vedpatwardhan 
I changed the error message in the latest commit